src/gti.jl

```diff
@@ -0,0 +1,98 @@
+function time_intervals_from_gtis(gtis::AbstractMatrix{<:Real}, segment_size::Real;
+                                  fraction_step::Real=1, epsilon::Real=1e-5)  
+    spectrum_start_times = Float64[]
+
+    gti_low = @view gtis[:,1]
+    gti_up = @view gtis[:,2]
+
+    for (g1,g2) in zip(gti_low,gti_up)
+        if g2 - g1 + epsilon < segment_size
+            continue
+        end
+
+        newtimes = range(g1, g2 - segment_size + epsilon, step=Float64(segment_size) * fraction_step)
+        append!(spectrum_start_times,newtimes)
+    end
+    return spectrum_start_times, spectrum_start_times .+ segment_size
+end
+
+function calculate_segment_bin_start(startbin::Integer, stopbin::Integer,
+                                     nbin::Integer; fraction_step::Real=1)
+    st = floor.(range(startbin, stopbin, step=Int64(nbin * fraction_step)))
```

Comment:
Unless you have a specific reason for using 64-bit integers, it's usually more natural to use `Int`, which is the integer type with the same number of bits as the current architecture (32-bit on 32-bit systems, 64-bit on 64-bit integers).  `Int` is also the type of integer literals in Julia
```suggestion
    st = floor.(range(startbin, stopbin, step=Int(nbin * fraction_step)))
```